Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -4111,7 +4111,8 @@ static void handleAnimationActions(Element& element, uint32_t action)
     // FIXME: Move settings from Frame to Frame and remove this check.
     auto& settings = frameView.frame().settings();
     LayoutRect documentRect = IntRect(frameView.scrollOrigin(), frameView.contentsSize());
-    auto layoutViewportSize = LocalFrameView::expandedLayoutViewportSize(frameView.baseLayoutViewportSize(), LayoutSize(documentRect.size()), settings.layoutViewportHeightExpansionFactor());
+    double heightExpansionFactor = m_allowsLayoutViewportHeightExpansion ? settings.layoutViewportHeightExpansionFactor() : 0;
+    auto layoutViewportSize = LocalFrameView::expandedLayoutViewportSize(frameView.baseLayoutViewportSize(), LayoutSize(documentRect.size()), heightExpansionFactor);
```

Comment:
Not new, but I think there's a bit of code duplication going on here with `WebPageProxyIOS`?

Reply:
Good point — these two codepaths share very similar logic, but they're on opposite sides of the process boundary (and thus read off of settings and preferences respectively). Most of the (easily shared) logic is already in the `LocalFrameView::expandedLayoutViewportSize` helper method.

Thanks for the review!